evaluate/repository_test.go

```diff
@@ -93,3 +93,77 @@ func TestRepository(t *testing.T) {
 		},
 	})
 }
+
+func TestTemporaryRepository(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		TestDataPath   string
+		RepositoryPath string
+
+		ExpectedTempPathRegex string
+		ExpectedErr           error
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			actualTemporaryRepositoryPath, actualErr := TemporaryRepository(tc.TestDataPath, tc.RepositoryPath)
+			defer func() {
+				assert.NoError(t, os.RemoveAll(actualTemporaryRepositoryPath))
+			}()
+
+			assert.Regexp(t, filepath.Join(os.TempDir(), tc.ExpectedTempPathRegex), actualTemporaryRepositoryPath)
+			assert.Equal(t, tc.ExpectedErr, actualErr)
+		})
+	}
+
+	validate(t, &testCase{
+		Name: "Create temporary path with git repository",
+
+		TestDataPath:   filepath.Join("..", "testdata"),
+		RepositoryPath: filepath.Join("golang", "plain"),
+
+		ExpectedTempPathRegex: `eval-dev-quality\d+/plain`,
+		ExpectedErr:           nil,
+	})
+}
+
+func TestResetTemporaryRepository(t *testing.T) {
+	type testCase struct {
+		Name string
+
+		TestDataPath   string
+		RepositoryPath string
+
+		ExpectedErr error
+	}
+
+	validate := func(t *testing.T, tc *testCase) {
+		t.Run(tc.Name, func(t *testing.T) {
+			temporaryRepositoryPath, err := TemporaryRepository(tc.TestDataPath, tc.RepositoryPath)
+			assert.NoError(t, err)
+			defer func() {
+				assert.NoError(t, os.RemoveAll(temporaryRepositoryPath))
+			}()
+
+			// Create a file
+			tmpFile := filepath.Join(temporaryRepositoryPath, "foo")
+			assert.NoError(t, os.WriteFile(tmpFile, []byte("foo"), 0600))
```

Comment:
Nitpick: this is somewhat testcase specific and would probably be better within the `testCase` instead of within `validate`.